File: src/main/java/org/eln/eln3/sim/ElectricalResistorHeatThermalLoad.java
package org.eln.eln3.sim;

import org.eln.eln3.sim.mna.component.Resistor;

public class ElectricalResistorHeatThermalLoad implements IProcess {

    Resistor electricalResistor;
    ThermalLoad thermalLoad;

    public ElectricalResistorHeatThermalLoad(Resistor electricalResistor, ThermalLoad thermalLoad) {
        this.electricalResistor = electricalResistor;
        this.thermalLoad = thermalLoad;
    }

    @Override
    public void process(double time) {
        thermalLoad.netThermalPowerAccumulator += electricalResistor.getPower();
    }
}

File: src/main/kotlin/org/eln/eln3/sim/process/thermal/ElectricalResistorHeatThermalLoad.kt
package org.eln.eln3.sim.process.thermal

import org.eln.eln3.sim.IProcess
import org.eln.eln3.sim.mna.component.Resistor
import org.eln.eln3.sim.thermal.ThermalLoad

class ElectricalResistorHeatThermalLoad(var electricalResistor: Resistor, var thermalLoad: ThermalLoad) : IProcess {
    override fun process(time: Double) {
        thermalLoad.netThermalPowerAccumulator = thermalLoad.netThermalPowerAccumulator + electricalResistor.getPower()
    }
}